doc/source/release-notes/Percona-Server-5.7.21-21.rst

```diff
@@ -0,0 +1,137 @@
+  :mysqlbug:`90238`).
+
+* ``mysqldump`` utility with ``--innodb-optimize-keys`` option was incorrectly
+  working with foreign keys on the same table, producing invalid SQL
+  statements. Bugs fixed :psbug:`1125` and :psbug:`3863`.
+
+* A fix of the mysqld startup script failed to detect jemalloc library
+  location for preloading, thus not starting on systemd based machines,
+  introduced in |Percona Server| :rn:`5.7.21-20`, was improved to take into
+  account previously created configuration file. Bug fixed :psbug:`3850`.
+
+* A fix introduced in |Percona Server| :rn:`5.7.10-2` which made busy server
+  not to prefer LRU flushing over flush list flushing too strongly was reverted
+  back because of the MT LRU flusher introduction, which removed LRU flushing
+  from the cleaner loop, Bug fixed :psbug:`3812`.
+
+* The possibility of a truncated bitmap file name was fixed in |InnoDB| logging
+  subsystem. Bug fixed :psbug:`3926`.
+
+* Non-instrumented functions were used in cached reads and writes of files in
+  fixed-size units. Bug fixed :psbug:`3937` (upstream :mysqlbug:`90264`).
+
+* A crash in the unsafe query warning checks with views took place for
+  ``UPDATE`` statement in case of statement binlogging format. Bug fixed
+  :psbug:`290`.
+
+MyRocks Changes
+===============
+
+* A re-implemented variable :variable:`rpl_skip_tx_api` allows to turn on simple
+  |RocksDB| write batches functionality, increasing replication performance
+  by the transaction api skip. Bug fixed :jirabug:`MYR-47`.
+
+* Unpack info bytes indicating the trailing space length in a padded varchar
+  were not read/skipped for value-less fields, causing data corruption and/or
+  server crash. Bug fixed :jirabug:`MYR-232`.
+
+TokuDB Changes
+===============
+
+* Two new variables introduced for the |TokuDB| fast updates feature,
+  :variable:`tokudb_enable_fast_update` and
+  :variable:`tokudb_enable_fast_upsert` should be now used instead of the 
+  ``NOAR`` keyword, which is now optional at compile time and off by default.
+  Bugs fixed :tdbbug:`63` and :tdbbug:`148`.
+
+* A set of compilation fixes as introduced to make |TokuDB| successfully
+  build in 8.0. Bugs fixed :tdbbug:`84`, :tdbbug:`85`, :tdbbug:`114`,
+  :tdbbug:`115`, :tdbbug:`118`, :tdbbug:`128`, :tdbbug:`139`, :tdbbug:`141`,
+  and :tdbbug:`172`.
+
+* Conditional compilation code dependent on version ID in the TokuDB tree was
+  separated and arranged to specific version branches. Bugs fixed
+  :tdbbug:`133`, :tdbbug:`134`, :tdbbug:`135`, and :tdbbug:`136`.
+
+* An additional code clean-up was made in bounds of 8.0 transition to remove
+  MariaDB-specific constructions. Bugs fixed :tdbbug:`180`, :tdbbug:`181`,
+  and :tdbbug:`182`.
+
+* Alter table comment caused |TokuDB| to rebuild the whole table, which is not
+  needed, as only FRM metadata should be changed. Bug fixed :tdbbug:`130`,
+  and :tdbbug:`137`.
+
+* Data race on the cache table pair attributes was fixed.
```

Comment:
No bug ref?

Reply:
This is my fault. The change came in as a PR from Rik Prohaska which I reviewed and merged straight away without a JIRA tracker.